### DISCUSSION TITLE
ListItem to Object Mapper

### DISCUSSION THREAD

**Discussion by Tanddant** [pnp/pnpcore#1231] [Ideas]
Me and @jeppesc11 have been playing with the idea of a making a tool to map that can map List Items to C# class

Something like this:
```
[SPList(listTitle: "EmployeeTasks")]
internal class EmployeeTaskModel {
    [SPField(internalName: "ID")]
    public string? Id { get; set; }

    [SPField(internalName: "EmployeeTask_Done")]
    public bool? Done { get; set; }

    [SPField(internalName: "EmployeeTask_Employeer")]
    public FieldLookupValue? Employeer { get; set; }
}
```

    ctx.Web.GetItems<EmployeeTaskModel>.AsRequested();

We have a small POC working here: https://github.com/jeppesc11/ListItemDynamicMapper

Is this something that has come up before? - or are there obvious downsides that we're not seeing? 

Some of the challenges that're not yet taken care of:
- Saving back to SharePoint
- Lookup fields (how should they be mapped) something like string for LookupValue, int for ID
- What's the ideal way it should work? - our thoughts have been something like this:

````
ctx.Web.GetItems<EmployeeTaskModel>.AsRequested();
ctx.Web.Lists.GetByTitle("EmployeeTasks").Items<EmployeeTaskModel>.AsRequested();
````

**Comment by jansenbe** [comment]
Really cool idea @Tanddant  and @jeppesc11!! Would love to see such an implementation in PnP Core SDK. As I'll be off from tomorrow evening until the middle of August I've no bandwidth to work on this anytime soon, however I suggest you iterate on your POC so it can serve as a basis for the future PR to get this included. Maybe @PaoloPia can help you during my break?
